airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalUtils.java

```diff
@@ -37,59 +37,118 @@
 
+@Slf4j
 public class TemporalUtils {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(TemporalUtils.class);
+  private static final Configs configs = new EnvConfigs();
+  private static final Duration WORKFLOW_EXECUTION_TTL = Duration.ofDays(configs.getTemporalRetentionInDays());
+  private static final Duration WAIT_INTERVAL = Duration.ofSeconds(2);
+  private static final Duration MAX_TIME_TO_CONNECT = Duration.ofMinutes(2);
+  private static final Duration WAIT_TIME_AFTER_CONNECT = Duration.ofSeconds(5);
+  private static final String HUMAN_READABLE_WORKFLOW_EXECUTION_TTL =
+      DurationFormatUtils.formatDurationWords(WORKFLOW_EXECUTION_TTL.toMillis(), true, true);
 
+  public static final String DEFAULT_NAMESPACE = "default";
   public static final Duration SEND_HEARTBEAT_INTERVAL = Duration.ofSeconds(10);
   public static final Duration HEARTBEAT_TIMEOUT = Duration.ofSeconds(30);
-
-  public static WorkflowServiceStubs createTemporalService(final String temporalHost) {
-    final WorkflowServiceStubsOptions options = WorkflowServiceStubsOptions.newBuilder()
-        .setTarget(temporalHost) // todo: move to EnvConfigs
-        .build();
-
-    return getTemporalClientWhenConnected(
-        Duration.ofSeconds(2),
-        Duration.ofMinutes(2),
-        Duration.ofSeconds(5),
-        () -> WorkflowServiceStubs.newInstance(options));
-  }
-
   public static final RetryOptions NO_RETRY = RetryOptions.newBuilder().setMaximumAttempts(1).build();
-
-  private static final Configs configs = new EnvConfigs();
   public static final RetryOptions RETRY = RetryOptions.newBuilder()
       .setMaximumAttempts(configs.getActivityNumberOfAttempt())
       .setInitialInterval(Duration.ofSeconds(configs.getDelayBetweenActivityAttempts()))
       .build();
 
-  public static final String DEFAULT_NAMESPACE = "default";
+  public static WorkflowServiceStubs createTemporalProductionService() {
+    if (configs.temporalCloudEnabled()) {
+      log.info("createTemporalProductionService chose Cloud...");
+      return createTemporalCloudService();
+    }
+    log.info("createTemporalProductionService chose Airbyte...");
+    final WorkflowServiceStubs temporalService = createTemporalAirbyteService();
+    configureTemporalAirbyteNamespace(temporalService);
+    return temporalService;
+  }
 
-  private static final Duration WORKFLOW_EXECUTION_TTL = Duration.ofDays(configs.getTemporalRetentionInDays());
-  private static final String HUMAN_READABLE_WORKFLOW_EXECUTION_TTL =
-      DurationFormatUtils.formatDurationWords(WORKFLOW_EXECUTION_TTL.toMillis(), true, true);
+  // TODO consider making this private after the Temporal Cloud migration
+  public static WorkflowServiceStubs createTemporalCloudService() {
+    return createTemporalCloudService(
+        configs.getTemporalCloudClientCert(),
+        configs.getTemporalCloudClientKey(),
+        configs.getTemporalCloudHost(),
+        configs.getTemporalCloudNamespace());
+  }
```

Comment:
same thought here on consolidating these methods

Reply:
Yep, in this case I think consolidation makes sense! I would have liked to do something similar for the `TemporalClient.get` method as you suggested, but the migrator needs to explicitly instantiate both types of clients, so for now we need two separate public methods there for the migrator to call.